sympy/printing/latex.py

```diff
@@ -446,9 +446,15 @@ def convert(expr):
             snumer = convert(numer)
             sdenom = convert(denom)
             ldenom = len(sdenom.split())
-            ratio = self._settings['long_frac_ratio']
+            if self._settings['long_frac_ratio'] == None:
+                ratio = len(expr.as_numer_denom()[0].args)
```

Comment:
Much better now. But still, I don't think making up a value for `ratio` is the way to go. If it's a setting that should not apply when None, then it should not apply (below in the code, where it is used). 

With this change, you still have `latex((x+y)/z)`  printing as "\frac{1}{x} \left(x + y\right)". Why this behavior, if we are okay with long fractions? That's still that ratio at work.  `latex((x+y)/(z+w))`  prints as `\\frac{x + y}{w + z}'.  

You were concerned about tests failing. Such things happen when the default setting of a parameter is changed. Tests should be reviewed and revised as necessary.